ledger/sandbox/src/main/scala/com/digitalasset/platform/index/Disclosure.scala

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2020 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.index
+
+import com.digitalasset.daml.lf.data.Ref.Party
+import com.digitalasset.daml.lf.data.Relation.Relation
+import com.digitalasset.daml.lf.engine.Blinding
+import com.digitalasset.daml.lf.transaction.Transaction.{NodeId, TContractId, Transaction, Value}
+import com.digitalasset.daml.lf.transaction.{GenTransaction, NodeInfo}
+
+object Disclosure {
+
+  def forFlatTransaction[Nid, Cid, Val](tx: GenTransaction[Nid, Cid, Val]): Relation[Nid, Party] =
+    tx.nodes.collect {
+      case (nodeId, c: NodeInfo.Create) =>
+        nodeId -> c.stakeholders
+      case (nodeId, e: NodeInfo.Exercise) if e.consuming =>
+        nodeId -> e.stakeholders
+    }
+
+  def forTransactionTree[Nid, Cid, Val](
+      disclosure: GenTransaction[Nid, Cid, Val] => Relation[Nid, Party])(
```

Comment:
Why not pass in the disclosure relation directly?

Reply:
I was hoping I could somehow make this testable by abstracting over this, but I guess I can undo this.

Reply:
Plus, initially I used this to test this in the current `TransactionConversion`.

Reply:
If yo don't intend to use it, then let's keep it to the minimum that's needed. 👍

Reply:
Addressed by https://github.com/digital-asset/daml/pull/4970/commits/f8e11794c14a26621f54a77274d02ab8afe56067